src/code/z_parameter.c

```diff
@@ -52,38 +332,750 @@
+        }
+    }
+}
+
+s32 Inventory_ReplaceItem(GlobalContext* globalCtx, u8 oldItem, u8 newItem) {
+    u8 i;
+
+    for (i = 0; i < 24; i++) {
+        if (gSaveContext.save.inventory.items[i] == oldItem) {
+            gSaveContext.save.inventory.items[i] = newItem;
+
+            for (i = 1; i < 4; i++) {
+                if (GET_CUR_FORM_BTN_ITEM(i) == oldItem) {
+                    SET_CUR_FORM_BTN_ITEM(i, newItem);
+                    Interface_LoadItemIconImpl(globalCtx, i);
+                    break;
+                }
+            }
+            return true;
+        }
+    }
+    return false;
+}
+
+void Inventory_UpdateDeitySwordEquip(GlobalContext* globalCtx) {
+    InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
+    u8 btn;
+
+    if (CUR_FORM == PLAYER_FORM_FIERCE_DEITY) {
+        interfaceCtx->unk_21C = 0;
+        interfaceCtx->unk_21E = 0;
+        if ((((gSaveContext.save.playerForm > 0) && (gSaveContext.save.playerForm < 4))
+                 ? 1
+                 : gSaveContext.save.playerForm >> 1) == 0) {
+            BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) = ITEM_SWORD_DEITY;
+        } else {
+            if (BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) == ITEM_SWORD_DEITY) {
+                if (GET_CUR_EQUIP_VALUE(EQUIP_SWORD) == 0) {
+                    BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) = ITEM_NONE;
+                } else {
+                    BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) =
+                        GET_CUR_EQUIP_VALUE(EQUIP_SWORD) + ITEM_SWORD_KOKIRI - 1;
+                }
+            }
+        }
+    }
+
+    for (btn = 0; btn < 1; btn++) {
+        if ((GET_CUR_FORM_BTN_ITEM(btn) != ITEM_NONE) && (GET_CUR_FORM_BTN_ITEM(btn) != ITEM_UNK_FD)) {
+            Interface_LoadItemIconImpl(globalCtx, btn);
+        }
+    }
+}
+
+s32 Inventory_HasEmptyBottle(void) {
+    s32 slot;
+
+    for (slot = SLOT_BOTTLE_1; slot <= SLOT_BOTTLE_6; slot++) {
+        if (gSaveContext.save.inventory.items[slot] == ITEM_BOTTLE) {
+            return true;
+        }
+    }
+    return false;
+}
+
+s32 Inventory_HasItemInBottle(u8 item) {
+    s32 slot;
+
+    for (slot = SLOT_BOTTLE_1; slot <= SLOT_BOTTLE_6; slot++) {
+        if (gSaveContext.save.inventory.items[slot] == item) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void Inventory_UpdateBottleItem(GlobalContext* globalCtx, u8 item, u8 btn) {
+    gSaveContext.save.inventory.items[GET_CUR_FORM_BTN_SLOT(btn)] = item;
+    SET_CUR_FORM_BTN_ITEM(btn, item);
+
+    Interface_LoadItemIconImpl(globalCtx, btn);
+
+    globalCtx->pauseCtx.cursorItem[0] = item;
+    gSaveContext.buttonStatus[btn] = BTN_ENABLED;
+
+    if (item == ITEM_HOT_SPRING_WATER) {
+        func_8010EBA0(60, GET_CUR_FORM_BTN_SLOT(btn) - SLOT_BOTTLE_1);
+    }
+}
+
+s32 Inventory_ConsumeFairy(GlobalContext* globalCtx) {
+    u8 bottleSlot = SLOT(ITEM_FAIRY);
+    u8 btn;
+    u8 i;
+
+    for (i = 0; i < 6; i++) {
```

Comment:
I suppose if you really wanted to you could write these in terms of differences of bottle slots.

Reply:
I was thinking about this, this loop over the 6 bottles appears a lot in parameter. Could we make a define like:
`#define BOTTLE_FIRST 1`
`#define BOTTLE_LAST 5` (and of course do `i <= 5`)

Or maybe use the existing ones like:
`for (i = SLOT_BOTTLE_1 - SLOT_BOTTLE_1; i <= SLOT_BOTTLE_6 - SLOT_BOTTLE_1; i++)`

but then it looks kinda ugly

Reply:
I think the latter is more what Anghelo has been using, but I agree that it makes sense to have the former... the interesting question is how many files would want it.

Reply:
So are you thinking:
```
#define BOTTLE_FIRST 0
#define BOTTLE_MAX 6
```
or
```
typedef enum {
    /* 0 */ BOTTLE_0,
    /* 1 */ BOTTLE_1,
    /* 2 */ BOTTLE_2,
    /* 3 */ BOTTLE_3,
    /* 4 */ BOTTLE_4,
    /* 5 */ BOTTLE_5,
    /* 6 */ BOTTLE_MAX,
} Bottles;
```
(Or even 1 - 6 instead of 0 -5)

Reply:
I doubt it treats each bottle individually ever, so the latter seems a bit redundant.